app/assets/javascripts/accordion.js

```diff
@@ -6,7 +8,205 @@ document.addEventListener('DOMContentLoaded', function (event) {
     for (i = 0; i < x.length; i++) {
         x[i].addEventListener("click", function () {
             var child = this.nextElementSibling;
-            child.classList.toggle("open");
+            if (child.className == "card-body open") {
+                child.className = "card-body"
+            } else {
+                child.className = "card-body open"
+            }
         })
     }
+
+
+    //initial check on load for existing works
+    var p = document.getElementById("publication_content_genre");
+    if (p != null) {
+
+        //article fields
+        var confName = document.getElementsByClassName("form-group string optional publication_conference_name");
+        var isbn = document.getElementsByClassName("form-group string optional publication_isbn");
+        var series = document.getElementsByClassName("form-group string optional publication_series_title");
+        var edition = document.getElementsByClassName("form-group string optional publication_edition");
+        var publisherVersion = document.getElementsByClassName("form-group select optional publication_publisher_version")[0].firstChild;
+        var parTitle = document.getElementsByClassName("form-group string optional publication_parent_title")[0].firstChild;
+
+        formSetup();
+
+        //get select element and watch if it changes
+        p.addEventListener("change", function () {
+            formSetup();
+        });
+
+        function formSetup() {
+            var selectedValue = p.selectedOptions[0].label;
+            switch (selectedValue) {
+                case 'Article': {
+                    resetForm();
+                    console.log("Chose: Article");
+                    confName[0].style.display = "none";
+                    isbn[0].style.display = "none";
+                    series[0].style.display = "none";
+                    edition[0].style.display = "none";
+                    publisherVersion.insertAdjacentHTML("afterend", '&nbsp;<span id="pubform-pubver" class="badge badge-info required-tag">required</span>');
+                    parTitle.insertAdjacentHTML("afterend", '&nbsp;<span id="pubform-partitle" class="badge badge-info required-tag">required</span>');
+                    break;
+                }
+
+                case 'Book': {
```

Comment:
Since all other cases involve the same instructions, could you use a `default` case that resets the form and console logs the selected value?